miniapps/meshing/minimal-surface.cpp

```diff
@@ -0,0 +1,1302 @@
+            QE[3*NEPF + el_offset][1] = y_off + NV1D*NV1D*iy + ix + 1;
+            QE[3*NEPF + el_offset][2] = y_off + NV1D*NV1D*(iy+1) + ix + 1;
+            QE[3*NEPF + el_offset][3] = y_off + NV1D*NV1D*(iy+1) + ix;
+            // z = 0
+            QE[4*NEPF + el_offset][0] = NV1D*iy + ix;
+            QE[4*NEPF + el_offset][1] = NV1D*iy + ix + 1;
+            QE[4*NEPF + el_offset][2] = NV1D*(iy+1) + ix + 1;
+            QE[4*NEPF + el_offset][3] = NV1D*(iy+1) + ix;
+            // z = 1
+            int z_off = NV1D*NV1D*(NV1D-1);
+            QE[5*NEPF + el_offset][0] = z_off + NV1D*iy + ix;
+            QE[5*NEPF + el_offset][1] = z_off + NV1D*iy + ix + 1;
+            QE[5*NEPF + el_offset][2] = z_off + NV1D*(iy+1) + ix + 1;
+            QE[5*NEPF + el_offset][3] = z_off + NV1D*(iy+1) + ix;
+         }
+      }
+      // Delete on x = 0 face
+      QE[0*NEPF + 1 + 2*(NV1D-1)][0] = -1;
+      QE[0*NEPF + 1 + 1*(NV1D-1)][0] = -1;
+      // Delete on x = 1 face
+      QE[1*NEPF + 1 + 2*(NV1D-1)][0] = -1;
+      QE[1*NEPF + 1 + 1*(NV1D-1)][0] = -1;
+      // Delete on y = 1 face
+      QE[3*NEPF + 1 + 0*(NV1D-1)][0] = -1;
+      QE[3*NEPF + 1 + 1*(NV1D-1)][0] = -1;
+      // Delete on z = 1 face
+      QE[5*NEPF + 0 + 1*(NV1D-1)][0] = -1;
+      QE[5*NEPF + 1 + 1*(NV1D-1)][0] = -1;
+      QE[5*NEPF + 2 + 1*(NV1D-1)][0] = -1;
+      // Delete on z = 0 face
+      QE[4*NEPF + 1 + 0*(NV1D-1)][0] = -1;
+      QE[4*NEPF + 1 + 1*(NV1D-1)][0] = -1;
+      QE[4*NEPF + 1 + 2*(NV1D-1)][0] = -1;
+      // Delete on y = 0 face
+      QE[2*NEPF + 1 + 0*(NV1D-1)][0] = -1;
+      QE[2*NEPF + 1 + 1*(NV1D-1)][0] = -1;
+      for (int j = 0; j < NV; j++) { AddVertex(QV[j]); }
+      for (int j = 0; j < NE; j++)
+      {
+         if (QE[j][0] < 0) { continue; }
+         AddQuad(QE[j], j+1);
+      }
+      RemoveUnusedVertices();
+      FinalizeQuadMesh(false, 0, true);
+      EnsureNodes();
+      FinalizeTopology();
+   }
+
+   void Snap() { SnapNodesToUnitSphere(); }
+};
+
+static int Problem0(Opt &opt)
+{
+   // Create our surface mesh from command line options
+   Surface *S = nullptr;
+   switch (opt.surface)
+   {
+      case 0: S = new MeshFromFile(opt); break;
+      case 1: S = new Catenoid(opt); break;
+      case 2: S = new Helicoid(opt); break;
+      case 3: S = new Enneper(opt); break;
+      case 4: S = new Hold(opt); break;
+      case 5: S = new Costa(opt); break;
+      case 6: S = new Shell(opt); break;
+      case 7: S = new Scherk(opt); break;
+      case 8: S = new FullPeach(opt); break;
+      case 9: S = new QuarterPeach(opt); break;
+      case 10: S = new SlottedSphere(opt); break;
+      default: MFEM_ABORT("Unknown surface (surface <= 10)!");
+   }
+   S->Create();
+   S->Solve();
+   delete S;
+   return 0;
+}
+
+// Problem 1: solve the Dirichlet problem for the minimal surface equation
+//            of the form z=f(x,y), using Picard iterations.
+static double u0(Vector &x) { return sin(3.0 * PI * (x[1] + x[0])); }
```

Comment:
I believe `FunctionCoefficients` based on this type of function have been deprecated.  The argument `const Vector &x` is preferred.

Reply:
I've added the `const`, right.

Reply:
`done`